File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        hmap = {} 
        res = 0
        for i in nums:
            if i not in hmap:
                hmap[i] = hmap.get(i+1, 0) + hmap.get(i-1, 0) + 1
                if hmap.get(i-1, -1) != -1: hmap[i - hmap[i-1]] = hmap[i]
                if hmap.get(i+1, -1) != -1: hmap[i + hmap[i+1]] = hmap[i]
                res = max(res, hmap[i])
        return res

File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int max = 0;
        int l = 0;
        int r = heights.size() - 1;
        while (l < r) {
            int h;
            if (heights[l] < heights[r]) h = heights[l];
            else h = heights[r];
            int a = (r-l) * h;
            if (a > max) max = a;
            if (heights[l] < heights[r]) {
                l++;
            } else r--;
        }
        return max;
    }
};
